src/main/java/com/squareup/kotlinpoet/TypeAliasSpec.kt

```diff
@@ -67,9 +71,12 @@ class TypeAliasSpec private constructor(builder: TypeAliasSpec.Builder) {
       require(isName(name)) { "not a valid name: $name" }
     }
 
-    fun visibility(modifier: KModifier) = apply {
-      require(modifier == KModifier.PUBLIC || modifier == KModifier.INTERNAL
-          || modifier == KModifier.PRIVATE) {
+    fun addModifiers(vararg modifiers: KModifier) = apply {
+      modifiers.forEach(this::addModifier)
+    }
+
+    private fun addModifier(modifier: KModifier) {
+      require(modifier in setOf(PUBLIC, INTERNAL, PRIVATE, IMPL)) {
```

Comment:
This idiom is incredibly inefficient, right? I wonder if there’s a better one that’s just as readable

Reply:
Right, in this case it will actually allocate a set, I went all in for brevity here. I vaguely remember though that there was a way to make it compile to a string of equality checks, or is it for ranges only? @JakeWharton do you know such thing? I'm also totally fine with going with separate checks here.

Reply:
When I was writing about seductive code, this is totally it. I love how well this reads, and yet I hate the performance consequences.

Reply:
Such fine-grained performance concern isn't needed in this specific library, but as a general concept I totally agree. This actually may be a great case for collection literals because the compiler can optimize it into a more efficient shape in the bytecode (if/elses, tableswitch, constant, something else?).

A quick hack would be manual overloads for 1..N to get:
```kotlin
inline fun <T> requireIn(
    subject: T,
    expected1: T,
    expected2: T,
    expected3: T,
    expected4: T,
    error: () -> String
) {
  require(subject != expected1
      && subject != expected2
      && subject != expected3
      && subject != expected4,
      error)
}
```

Reply:
Not as readable on the usage site though: `requireIn(modifier, PUBLIC, PRIVATE)` vs `require(modifier in setOf(PUBLIC, PRIVATE))`. An optimization would be to have an `internal val SUPPORTED_MODIFIERS` in the companion object, if we want to optimize.

Reply:
I _might_ do that for Android depending on call frequency but not for the JVM.

Reply:
There's other language levers we can pull to get the API into something equally as readable:
```kotlin
inline fun <T> T.isOneOf(
    expected1: T,
    expected2: T,
    expected3: T,
    expected4: T
) = this != expected1
    && this != expected2
    && this != expected3
    && this != expected4
```
```kotlin
require(modifier.isOneOf(PUBLIC, INTERNAL, PRIVATE, IMPL)) { ... }
```

Reply:
Nice!